### PR TITLE
Improving tests + adding CI images

### PR DESCRIPTION
**Pull request by robscott**
This branch makes a number of CI related updates:
- Adds CI images for automatic builds
- Adds tests for `pip` and `npm` install
- Adds support to `install-rok8s-requirements` for `apk` and `brew`
- Moves from CircleCI 1 to 2

This is mostly ready, but we still need to figure out how/where these images should be pushed.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+      - run:
+          name: Install dependencies
+          command: |
+            apt-get update
+            apt-get install -y shellcheck
+
+      - run:
+          name: Test pip install and ensure scripts are added to path
+          command: |
+            pip install git+https://github.com/reactiveops/rok8s-scripts.git@$CIRCLE_SHA1
```
Comment (ejether) [inline .circleci/config.yml]: Would it be better to do `pip install .` so we are testing the installation of the already pulled files?

Comment (robscott) [reply .circleci/config.yml]: Somehow I didn't realize that was an option - good idea!

**Comment by robscott** [comment]
@ejether I've removed the root level Dockerfiles now in favor of the `ci-images` directory. The images are now pushing to Quay from Circle automatically, which I think may be ideal here. Unfortunately Quay's UI to set up build triggers would turn into a fairly cumbersome process when we add new images here. Do you have any ideas for how I could test the release flow with Circle without actually breaking all the things?

**Comment by ejether** [comment]
I'm not sure I follow what you mean?

**Comment by robscott** [comment]
@ejether Release testing seemed to work well for me, I think this PR may be ready now.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,139 @@
+version: 2
+
+references:
```
Comment (ejether) [inline .circleci/config.yml]: Nice!

File: bin/install-rok8s-requirements [hunk 3]
```diff
@@ -24,12 +24,20 @@
 ROK8S_INSTALL_PATH="${ROK8S_INSTALL_PATH:-/usr/local/bin}"
 mkdir -p "${ROK8S_INSTALL_PATH}"
 
-PKG_MANAGER=$( command -v yum || command -v apt-get ) || echo "Neither yum nor apt-get found"
-KUBECTL_VERSION="${KUBECTL_VERSION:-v1.8.5}"
+PKG_MANAGER=$( command -v yum || command -v apt-get || command -v apk || command -v brew ) || echo "Supported package manager not found"
+
+if [[ $PKG_MANAGER = *"apk"* ]]; then
```
Comment (ejether) [inline bin/install-rok8s-requirements]: Awesome!

**Comment by robscott** [comment]
That's a great question. I'd assumed that the idea was to have different languages available should someone want to run application tests in CI (outside of the Docker containers they're building). I'm not sure if that's the case with the Node 7 one though. It would probably be hard to keep up with everything, but at least having Debian and Alpine images might make sense? We could also have Node and Ruby if we want to support common languages here.

**Comment by ejether** [comment]
I'm sure the node image was original used because it was installed from NPM instead of copied in directly. 

If we have an image per language more realistically, per language version and potentially per rok8s-version that could get to be a lot. Lets worry about that when we get there though. This all looks good to me.

**Pull request merged**